vertx-sql-client/src/main/java/io/vertx/sqlclient/data/NullValue.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2011-2021 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.sqlclient.data;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.codegen.annotations.GenIgnore;
+import io.vertx.core.VertxException;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+
+import java.math.BigDecimal;
+import java.time.*;
+import java.time.temporal.Temporal;
+import java.util.UUID;
+
+/**
+ * Instances of this class indicate the type associated with a {@code NULL} column value.
+ * This is useful when infering database types from {@link io.vertx.sqlclient.Tuple} items.
+ */
+@DataObject
+public class NullValue {
```

Comment:
I think we actually should simply drop `@DataObject` and make it a plain class to avoid restriction of data objects, like we did for `Numeric`